src/transports/ws/ws.c

```diff
@@ -148,81 +158,3 @@ static int nn_ws_optset_getopt (struct nn_optset *self, int option,
         return -ENOPROTOOPT;
     }
 }
-
-int nn_ws_send (int s, const void *msg, size_t len, uint8_t msg_type, int flags)
-{
-    int rc;
-    struct nn_iovec iov;
-    struct nn_msghdr hdr;
-    struct nn_cmsghdr *cmsg;
-    size_t cmsgsz;
-
-    iov.iov_base = (void*) msg;
-    iov.iov_len = len;
-    
-    cmsgsz = NN_CMSG_SPACE (sizeof (msg_type));
-    cmsg = nn_allocmsg (cmsgsz, 0);
-    if (cmsg == NULL)
-        return -1;
-
-    cmsg->cmsg_level = NN_WS;
-    cmsg->cmsg_type = NN_WS_HDR_OPCODE;
-    cmsg->cmsg_len = NN_CMSG_LEN (sizeof (msg_type));
-    memcpy (NN_CMSG_DATA (cmsg), &msg_type, sizeof (msg_type));
-
-    hdr.msg_iov = &iov;
-    hdr.msg_iovlen = 1;
-    hdr.msg_control = &cmsg;
-    hdr.msg_controllen = NN_MSG;
-
-    rc = nn_sendmsg (s, &hdr, flags);
-
-    return rc;
-}
-
-int nn_ws_recv (int s, void *msg, size_t len, uint8_t *msg_type, int flags)
-{
-    struct nn_iovec iov;
-    struct nn_msghdr hdr;
-    struct nn_cmsghdr *cmsg;
-    void *cmsg_buf;
-    int rc;
-
-    iov.iov_base = msg;
-    iov.iov_len = len;
-
-    hdr.msg_iov = &iov;
-    hdr.msg_iovlen = 1;
-    hdr.msg_control = &cmsg_buf;
-    hdr.msg_controllen = NN_MSG;
-
-    rc = nn_recvmsg (s, &hdr, flags);
-    if (rc < 0)
-        return rc;
-
-    /* Find WebSocket opcode ancillary property. */
-    cmsg = NN_CMSG_FIRSTHDR (&hdr);
-    while (cmsg) {
-        if (cmsg->cmsg_level == NN_WS && cmsg->cmsg_type == NN_WS_HDR_OPCODE) {
-            *msg_type = *(uint8_t *) NN_CMSG_DATA (cmsg);
-            break;
-        }
-        cmsg = NN_CMSG_NXTHDR (&hdr, cmsg);
-    }
-
-    /*  WebSocket transport should always report this header. */
-    nn_assert (cmsg);
-
-    /*  WebSocket transport should always reassemble fragmented messages. */
-    nn_assert (*msg_type & NN_SWS_FRAME_BITMASK_FIN);
-
-    /*  Return only the message type (opcode). */
-    if (*msg_type == (NN_WS_MSG_TYPE_GONE | NN_SWS_FRAME_BITMASK_FIN))
-        *msg_type = NN_WS_MSG_TYPE_GONE;
-    else
-        *msg_type &= NN_SWS_FRAME_BITMASK_OPCODE;
-
-    nn_freemsg (cmsg_buf);
-
-    return rc;
-}
```

Comment:
I agree removing these websocket-specific exported helper methods. It's much better to set message type as socket-level option :-)